### PR TITLE
Refresh Service Fabric cmdlets

### PR DESCRIPTION
**Pull request by juhacket**
- Updated ARM templates
  - Failed operations no longer rollback
  - Add-AzureRmServiceFabricNodeType
    - VMs default to managed disks
    - Existing VMSS subnet used
    - All operations are idempotent
  - Remove-AzureRmServiceFabricNodeType cleans up partially created VMSS and/or cluster node types
  - Added new unit tests
    - Test-NewAzureRmServiceFabricCluster
    - Test-AddAzureRmServiceFabricNode
    - Test-RemoveAzureRmServiceFabricNode
    - Test-RemoveAzureRmServiceFabricNodeType
  - Updated Test-UpdateAzureRmServiceFabricReliability to test Silver to Bronze reliability downgrade
  - Updated Test-UpdateAzureRmServiceFabricDurability to test Bronze to Silver durability upgrade
  - Fixed output of PSCluster object for complex property types
  - Updated to Microsoft.Azure.Management.Compute.17.3.0
  - Updated to Microsoft.Rest.ClientRuntime.Azure.TestFramework.1.7.2
  - Updated to Microsoft.Azure.Management.Network.15.1.0-preview
  - Created resource strings for exception messages
  - Moved test VM image to Windows Server 2016 with Containers
  - Cleaned up duplicate references in unit test project
  - Isolated SFRP client calls
  - Removed redundant cluster read operations

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#making-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/RemoveAzureRmServiceFabricNodeType.cs [hunk 1]
```diff
@@ -45,58 +48,82 @@ public class RemoveAzureRmServiceFabricNodeType : ServiceFabricNodeTypeCmdletBas
...
+                    {
+                        MaxPercentDeltaUnhealthyApplications = 0,
+                        MaxPercentDeltaUnhealthyNodes = 0,
+                        MaxPercentUpgradeDomainDeltaUnhealthyNodes = 0
 
-            if (ShouldProcess(target: this.NodeType, action: string.Format("Remove a nodetype {0} ", this.NodeType)))
-            {
-                cluster = RemoveNodeTypeFromSfrp();
-                this.ComputeClient.VirtualMachineScaleSets.Delete(this.ResourceGroupName, this.NodeType);
+                        // TODO: Enable with next SDK version
```
Comment (cormacpayne) [inline src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/RemoveAzureRmServiceFabricNodeType.cs]: @juhacket nit: can we remove this commented-out section?

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricClusterCmdlet.cs [hunk 2]
```diff
@@ -39,6 +42,77 @@ public abstract class ServiceFabricClusterCmdlet : ServiceFabricCmdletBase
...
+                            {
+                                WriteVerboseWithTimestamp(
+                                    string.Format(
+                                        ServiceFabricProperties.Resources.ClusterStateVerbose,
+                                        c.ClusterState));
+                            }
+                        }
+                    }
+
+                    Thread.Sleep(TimeSpan.FromSeconds(WriteVerboseIntervalInSec));
```
Comment (cormacpayne) [inline src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricClusterCmdlet.cs]: @juhacket what is this needed for?

Comment (juhacket) [reply src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricClusterCmdlet.cs]: This is required for us to change certain properties of the cluster resource. The current API only supports a subset of these properties (SendPatchRequest). We are going to expose these properties to the patch request in the next SDK update. We will merge SendPutRequest into SendPatchRequest in that change as well.

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands.ServiceFabric.csproj [hunk 3]
```diff
@@ -173,6 +171,12 @@
     <Reference Include="Microsoft.Azure.Management.Storage, Version=6.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Storage.6.5.0-preview\lib\net452\Microsoft.Azure.Management.Storage.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
[the remark this answers is not in the document]
Comment (cormacpayne) [reply src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands.ServiceFabric.csproj]: @juhacket ping on this comment

[the remark this answers is not in the document]
Comment (juhacket) [reply src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands.ServiceFabric.csproj]: @cormacpayne  Rebase issue...

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricClusterCmdlet.cs [hunk 4]
```diff
@@ -22,13 +22,16 @@
...
 using Microsoft.Azure.Management.Compute;
+using Microsoft.Azure.Management.Compute.Models;
 using Microsoft.Azure.Management.ServiceFabric;
 using Microsoft.Azure.Management.ServiceFabric.Models;
 using Newtonsoft.Json.Linq;
 using ServiceFabricProperties = Microsoft.Azure.Commands.ServiceFabric.Properties;
 
 namespace Microsoft.Azure.Commands.ServiceFabric.Commands
 {
+    using System.Globalization;
```
Comment (cormacpayne) [inline src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricClusterCmdlet.cs]: @juhacket nit: would you mind moving this to be with all of the other usings?

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricCmdletBase.cs [hunk 5]
```diff
@@ -47,6 +48,8 @@
 
 namespace Microsoft.Azure.Commands.ServiceFabric.Commands
 {
+    using Microsoft.Rest;
```
Comment (cormacpayne) [inline src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricCmdletBase.cs]: @juhacket nit: same comment about moving this to be with the other usings

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricNodeTypeCmdletBase.cs [hunk 6]
```diff
@@ -27,6 +27,8 @@
 
 namespace Microsoft.Azure.Commands.ServiceFabric.Commands
 {
+    using Microsoft.Azure.Management.ServiceFabric.Models;
```
Comment (cormacpayne) [inline src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricNodeTypeCmdletBase.cs]: @juhacket nit: same comment about moving this using

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricSettingsCmdletBase.cs [hunk 7]
```diff
@@ -22,6 +22,8 @@
 
 namespace Microsoft.Azure.Commands.ServiceFabric.Commands
 {
+    using System.Linq;
```
Comment (cormacpayne) [inline src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricSettingsCmdletBase.cs]: @juhacket nit: same comment about moving this using

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Models/PSCluster.cs [hunk 8]
```diff
@@ -20,6 +20,8 @@
 
 namespace Microsoft.Azure.Commands.ServiceFabric.Models
 {
+    using System.Reflection;
```
Comment (cormacpayne) [inline src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Models/PSCluster.cs]: @juhacket nit: same comment about moving this using

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/help/New-AzureRmServiceFabricCluster.md [hunk 9]
```diff
@@ -61,9 +61,9 @@ $clusterloc="SouthCentralUS"
 $subname="{0}.{1}.cloudapp.azure.com" -f $RGname, $clusterloc
```
Comment (cormacpayne) [inline src/ResourceManager/ServiceFabric/Commands.ServiceFabric/help/New-AzureRmServiceFabricCluster.md]: @juhacket a note about the markdown files in general: it looks like the markdown files weren't regenerated after some public API changes were made across a few cmdlets. Would you mind refreshing them?

Comment (juhacket) [reply src/ResourceManager/ServiceFabric/Commands.ServiceFabric/help/New-AzureRmServiceFabricCluster.md]: Regenerated the help.

**Comment by juhacket** [comment]
@cormacpayne Fixed all comments.

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/packages.config [hunk 10]
```diff
@@ -2,9 +2,9 @@
 <packages>
   <package id="Microsoft.Azure.KeyVault" version="2.3.0-preview" targetFramework="net452" />
   <package id="Microsoft.Azure.KeyVault.WebKey" version="2.0.6" targetFramework="net452" />
-  <package id="Microsoft.Azure.Management.Compute" version="14.0.0-prerelease" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.Compute" version="17.3.0" targetFramework="net452" />
   <package id="Microsoft.Azure.Management.KeyVault" version="2.3.0-preview" targetFramework="net452" />
-  <package id="Microsoft.Azure.Management.Network" version="8.1.0-preview" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.Network" version="15.1.0-preview" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/ServiceFabric/Commands.ServiceFabric/packages.config]: @juhacket something that doesn't need to be addressed now, but we should look at in an upcoming sprint: we need to remove ServiceFabric's dependency on other services' management libraries. We can do this by (1) evaluating whether or not ServiceFabric needs the management library and (2) generating pieces of each library that ServiceFabric (and potentially other services) consume in their module and put that in `src/Common` along with other generated libraries.

Comment (cormacpayne) [reply src/ResourceManager/ServiceFabric/Commands.ServiceFabric/packages.config]: @juhacket here's the corresponding issue to track this work: https://github.com/Azure/azure-powershell/issues/5365

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/528/

**Comment by juhacket** [comment]
@cormacpayne @maddieclayton Updated change log.

**Pull request merged**